service/src/main/java/com/commercetools/pspadapter/payone/ScheduledJobFactory.java

```diff
@@ -17,28 +18,120 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(IntegrationService.class);
 
-    private static final int DELAY_START_IN_SECONDS = 10;
+    static final int DELAY_START_IN_SECONDS = 5;
 
-    public static void createScheduledJob(
+    private static final DateFormat DATE_FORMATTER = new SimpleDateFormat("dd/MM/yyyy HH:mm:ss");
+
+    private final Scheduler scheduler;
+
+    /**
+     * How many items are scheduled but not started yet.
+     */
+    private int scheduledItemsCount = 0;
+
+    /**
+     * Single listener for started scheduled items. This instance is added to {@code scheduler#getListenerManager()}
+     * if {@code allScheduledItemsStarted} is set.
+     */
+    private SchedulerListenerSupport schedulerStartedListener;
+
+    /**
+     * Callback to call when all scheduled jobs are started (e.g. when {@code scheduledItemsCount} becomes 0.
+     */
+    private Runnable allScheduledItemsStarted;
+
+    public ScheduledJobFactory() throws SchedulerException {
+        scheduler = new StdSchedulerFactory().getScheduler();
+    }
+
+    public JobDetail createScheduledJob(
             final CronScheduleBuilder cronScheduleBuilder,
             final Class<? extends ScheduledJob> jobClass,
-            final IntegrationService integrationService,
-            final PaymentDispatcher paymentDispatcher) throws SchedulerException {
+            final IntegrationService integrationService) throws SchedulerException {
 
         JobDataMap dataMap = new JobDataMap();
-        dataMap.put(ScheduledJob.SERVICE_KEY, integrationService);
-        dataMap.put(ScheduledJob.DISPATCHER_KEY, paymentDispatcher);
-        Scheduler scheduler = new StdSchedulerFactory().getScheduler();
+        dataMap.put(ScheduledJob.INTEGRATION_SERVICE, integrationService);
+
         Trigger trigger = TriggerBuilder
                 .newTrigger()
                 .withSchedule(cronScheduleBuilder)
                 .usingJobData(dataMap)
                 .build();
 
+        scheduledItemsCount++;
         scheduler.startDelayed(DELAY_START_IN_SECONDS);
-        Date date = scheduler.scheduleJob(JobBuilder.newJob(jobClass).build(), trigger);
-        DateFormat df = new SimpleDateFormat("dd/MM/yyyy HH:mm:ss");
 
-        LOG.info("Starting scheduled job '{}' at {}", jobClass.getSimpleName(), df.format(date));
+        JobDetail jobDetail = JobBuilder.newJob(jobClass).build();
+        Date date = scheduler.scheduleJob(jobDetail, trigger);
+
+        LOG.info("Starting scheduled job '{}' at {}", jobClass.getSimpleName(), DATE_FORMATTER.format(date));
```

Comment:
this log is service wide, not tenant specific (we run one scheduled job, but it processes sequentially all tenants)